[Nadine Chiun] iP

bloodofme opened the pull request:
# ChatCat

> Need a program written by java to help you keep track of your tasks ? **ChatCat** is the perfect program for you! :smile:

## How to download

1. Download the jar file by clicking on this [link](https://github.com/bloodofme/ip/releases)
2. Run commands `Java -jar ChatCat.jar`
3. Add your tasks 

## Why download it ?

- [x] Free
- [x] Easy to use

## What are ChatCat's features ?

- List down tasks
- Flag tasks as 3 different types of tasks : "Todo", "Event" and "Deadline"
- Mark and unmark tasks
- Delete tasks


FYI, this is the welcome message function from the source code :)
   ```ruby
            public void showWelcomeSMS() {
                String welcome = "Hello! I'm ChatCat\n"
                        + "What can I do for you?\n";
                System.out.println(welcome);
            }
   ```

Commit:
Add Gradle support

bloodofme committed:
completed Level 1 of tasks

bloodofme committed:
completed level 2 tasks

bloodofme committed:
completed level 3 tasks

bloodofme committed:
completed level 4

bloodofme committed:
successfully tested ChatCat

bloodofme committed:
Completed Level 5 tasks

bloodofme committed:
completed level 6

bloodofme committed:
Complete level 7

bloodofme committed:
complete level 8

bloodofme committed:
Merge branch 'branch-Level-8'

bloodofme committed:
Complete A-MoreOOP

bloodofme committed:
Complete A-Packages

bloodofme committed:
complete JUnit

bloodofme committed:
Complete A-Jar task

bloodofme committed:
Complete JavaDocs Task

bloodofme committed:
Complete A-JavaDoc task

bloodofme committed:
Complete A-CodingStandard task

bloodofme committed:
Complete Level-9 task

bloodofme committed:
Merge branch 'branch-A-CodingStandard'

bloodofme committed:
Merge branch 'branch-Level-9'

justinyjt commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+import dukeexception.DukeException;
+
+public class TaskList {
+    ArrayList<Task> Tasks = new ArrayList<>();
```
Would it be better to use access control specifiers for your class attributes? Most of your classes do no specify this 😅  Also, your variable name for ArrayList<Task> should start with a small letter 't' 👍

justinyjt commented on src/main/java/util/UI.java:
```diff
@@ -0,0 +1,18 @@
+/**
+ * Handles outputs to interact with user
+ */
+public class UI {
```
Should the classname be Ui instead (Pascal Case) even though UI is an acronym? ([source](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Abbreviations%20and%20acronyms%20should%20not%20be%20uppercase%20when%20used%20as%20a%20part%20of%20a%20name))

justinyjt commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,35 @@
+    private String task;
+    private boolean isDone;
+    private String Symbol;
+    private final String doneSymbol = "[X]";
```
If you intend to declare a constant, would it be better to declare it as a static final as it is a class level variable? Also, the convention for naming constants is fully uppercase letters with underscores to separate the words ([source](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Constant%20names%20must%20be%20all%20uppercase%20using%20underscore%20to%20separate%20words.)).

justinyjt reviewed (commented):
Overall, I feel that your code is well written and easy to read with proper indentations to make it neat and organised. There were no major violations of Java coding standards as well. 👍

bloodofme committed:
Solved write to file issue

bloodofme committed:
Merge branch 'add-gradle-support'

bloodofme committed:
merge gradle branch to master

file:///Users/bloodofme/Documents/uni/cs2103T/ip/.gradle/6.2/fileChanges/last-build.bin

aachia commented on src/main/java/chatcat/tasks/Task.java:
```diff
@@ -0,0 +1,63 @@
+    /**
+     * Unmarks a task {@code Task} as not completed.
+     */
+    public void setUnDone() {
```
```suggestion
    public void setUndone() {
```
Good naming of methods by using verbs here! Is it necessary to capitalize it in this manner though, since undone is a word itself.

aachia commented on src/main/java/chatcat/tasks/Task.java:
```diff
@@ -0,0 +1,63 @@
+     *
+     * @return true if task contains a keyword, else returns false.
+     */
+    public boolean containsKeyWord(String str) {
```
```suggestion
    public boolean containsKeyword(String str) {
```
Same concern with my previous comment where I think capitalising might not be necessary but otherwise good naming of the method!

aachia commented on src/main/java/chatcat/tasklist/TaskList.java:
```diff
@@ -0,0 +1,212 @@
+ * Allows clients to unmark, mark, add and delete tasks. Allows users to view task list.
+ */
+public class TaskList {
+    ArrayList<Task> Tasks = new ArrayList<>();
```
```suggestion
    ArrayList<Task> tasks = new ArrayList<>();
```
Great job naming the arraylist in plural form since its a list of tasks but variables should not start with a capitalization.

aachia reviewed (commented):
Overall a job well done and I found it easy to read with the good naming conventions and headers written. There were only a few minor code violations that I could find. Keep it up and all the best!

bloodofme committed:
Complete GUI task

bloodofme committed:
Complete GUI task

bloodofme committed:
Minor Coding Standard Improvement

bloodofme committed:
Encapsulate Commands

bloodofme committed:
Create new jar file

bloodofme committed:
Add assertions

Add assertions to ChatCat

Users are able to insert invalid inputs

Adding assertions would improve catching of invalid inputs placed by the
user

Assertions Added in,
* MarkCommand: index inserted by user has to be less than number of
tasks added
* UnmarkCommand: index inserted by user has to be less than number of
tasks added
* DeleteCommand: index inserted by user has to be less than number of
tasks added
* ListTaskCommand: list should not be empty

bloodofme committed:
Merge pull request #1 from bloodofme/branch-A-Assertions

Add assertions

bloodofme committed:
Improve coding standard

Improve coding standards

This is to improve the overall readibility of the code.

ChatCat has been reviewed and areas where coding standards can be
improve has been changed.

bloodofme committed:
Merge branch 'master' into branch-A-CodingStandard

bloodofme committed:
Merge pull request #2 from bloodofme/branch-A-CodingStandard

Improve coding standard

bloodofme committed:
Improve Code Quality

Improve coding quality

This is to improve the overall readibility of the code.

ChatCat has been reviewed and areas where coding standards can be
improve has been changed.

bloodofme committed:
Merge pull request #3 from bloodofme/branch-A-CodeQuality

Improve Code Quality

bloodofme committed:
Add duplicate task capture extension

bloodofme committed:
Merge branch 'master' of https://github.com/bloodofme/ip

bloodofme committed:
Check ChatCat Scripts

bloodofme committed:
Set theme jekyll-theme-slate

bloodofme committed:
Update README

bloodofme committed:
Merge branch 'master' of https://github.com/bloodofme/ip

bloodofme committed:
Update README.md

bloodofme committed:
Update README.md

bloodofme committed:
Edit README

bloodofme committed:
Merge branch 'master' of https://github.com/bloodofme/ip

bloodofme committed:
Updated README

bloodofme committed:
Update README.md

bloodofme committed:
Change file name from images to assets

bloodofme committed:
update image directory

bloodofme committed:
Merge branch 'master' of https://github.com/bloodofme/ip

bloodofme committed:
Change location of Ui image

bloodofme committed:
Add bye description in README

bloodofme committed:
Create Jar file, perform soft test on jar file
